recovery/index.html

```diff
@@ -262,6 +262,64 @@ <h2>Message log</h2>
         return bytes;
       }
 
+      /**
+       * Decodes a base58check-encoded string into a buffer
+       * This function throws an error when the string is too small, doesn't have the proper checksum, or contains invalid characters.
+       * Inspired by https://gist.github.com/diafygi/90a3e80ca1c2793220e5/
+       * @param {string} s
+       * @return {Uint8Array}
+       */
+      async function base58checkDecode(s) {
+        if (s.length < 5) {
+          throw new Error(`cannot base58-decode a string of length < 5 (found length ${s.length})`)
+        }
+
+        // See https://en.bitcoin.it/wiki/Base58Check_encoding
+        var alphabet = "123456789ABCDEFGHJKLMNPQRSTUVWXYZabcdefghijkmnopqrstuvwxyz";
+        var decoded = BigInt(0);
+        var decodedBytes = [];
+        var result = [];
+        for (var i = 0; i < s.length; i++) {
+          if (alphabet.indexOf(s[i]) === -1) {
+            throw new Error(`cannot base58-decode: ${s[i]} isn't a valid character`)
+          }
+          var carry = alphabet.indexOf(s[i]);
+
+          // If the current base58 digit is 0, append a 0 byte.
+          // "result.length == i" can only happen if we have not seen non-zero bytes so far
+          if (carry == 0 && result.length === i) {
+            result.push(0);
+          }
+
+          var j = 0;
+          while (j < decodedBytes.length || carry > 0) {
+            var currentByte = decodedBytes[j];
+            // shift the current byte 58 units and add the carry amount (or just add the carry amount if this is a new byte)
+            currentByte = currentByte ? currentByte * 58 + carry : carry;
```

Comment:
when would `currentByte` be falsey?

Reply:
When its value is `undefined` (`decodedBytes[j]` can be `undefined`):
```
> bytes = [] 
> bytes[0]
undefined
```

Reply:
I've reworked the code to make this clearer.